app/models/user.py

```diff
@@ -1,22 +1,24 @@
-from app.models import *
+from flask_login import UserMixin
+from sqlalchemy import Boolean, Column, Enum, ForeignKey, Integer, String
+from sqlalchemy.orm import relationship
 
+from app.models import Base
+from app.models.robots import UserRobotBase
 
-class User(Base, UserMixin):
+
+class User(UserRobotBase, UserMixin):
     __tablename__ = "user"
 
-    id = Column(Integer, primary_key=True, autoincrement=True)
     username = Column(String(20), nullable=False)
-    name = Column(String(20), nullable=False)
     lastname = Column(String(40), nullable=False)
     email = Column(String(120), nullable=False)
-    image_file = Column(String(240), nullable=False, default="default.jpg")
     password = Column(String(120), nullable=False)
-    location = Column(Geography(geometry_type="POINT", srid=4326))
-    domicile_geolocation = Column(Geography(geometry_type="POINT", srid=4326))
     is_admin = Column(Boolean, default=False, nullable=False)
-    messages = relationship("Messages", backref="author", lazy=True)
+    chatrooms = relationship("ChatRoom", back_populates="user", lazy=True)
     profile = relationship("Profile", uselist=False, back_populates="user")
     user_criteria = relationship("UserCriteria", uselist=False, back_populates="user")
+    user_matches_1 = relationship("UserMatches", back_populates="user")
+    user_matches = relationship("RobotMatches", back_populates="user")

```

Comment:
Are you sure you need to have matches here? Wouldn't it be enough to just have chat rooms 🤔

Reply:
If there are always 2 members in the chat rooms it means that every chat room is in a 1:many relation with the robot and a 1:many relation with the user. So a chat room can represent a match.